src/main/java/io/anserini/util/MapCollections.java

```diff
@@ -0,0 +1,247 @@
+/**
+ * Anserini: An information retrieval toolkit built on Lucene
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.anserini.util;
+
+
+import io.anserini.collection.BaseFileSegment;
+import io.anserini.collection.DocumentCollection;
+import io.anserini.collection.SegmentProvider;
+import io.anserini.collection.SourceDocument;
+
+import io.anserini.util.mapper.CountDocumentMapper;
+import io.anserini.util.mapper.DocumentMapper;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.kohsuke.args4j.CmdLineException;
+import org.kohsuke.args4j.CmdLineParser;
+import org.kohsuke.args4j.OptionHandlerFilter;
+import org.kohsuke.args4j.ParserProperties;
+import org.kohsuke.args4j.Option;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+public final class MapCollections {
+  private static final Logger LOG = LogManager.getLogger(MapCollections.class);
+
+  public static final class Args {
+
+    // required arguments
+
+    @Option(name = "-input", metaVar = "[Directory]", required = true, usage = "collection directory")
+    public String input;
+
+    @Option(name = "-threads", metaVar = "[Number]", required = true, usage = "Number of Threads")
+    public int threads;
+
+    @Option(name = "-collection", required = true, usage = "collection class in io.anserini.collection")
+    public String collectionClass;
+
+    @Option(name = "-generator", required = true, usage = "document generator in io.anserini.index.generator")
+    public String generatorClass;
+
+    @Option(name = "-mapper", required = true, usage = "mapper class in io.anserini.util.mapper")
+    public String mapperClass;
+
+    // optional arguments
+
+    @Option(name = "-output", metaVar = "[Path]", usage = "output path")
+    public String output;
```

Comment:
If I'm not mistaken, this isn't used anywhere? (Other than logging the variable?)

Reply:
Since developers can build their own mapper, I think we should allow them to write output to somewhere other than stdout? Providing this option will be useful to them...